### ISSUE TITLE
RGAN and RCGAN implementation

### ISSUE THREAD

**Issue by dsevero** [torchgan/torchgan#110]
I implemented RGAN and RCGAN from [1]. Would it make sense to add it to `torchgan.models`?

[1] [Esteban, Cristóbal, Stephanie L. Hyland, and Gunnar Rätsch. "Real-valued (medical) time series generation with recurrent conditional gans." arXiv preprint arXiv:1706.02633 (2017).](https://arxiv.org/abs/1706.02633)

**Comment by avik-pal** [comment]
@dsevero It seems like an interesting model to have. Go ahead an open a PR for that.

**Comment by avik-pal** [comment]
Yes seems a valid thing to do. Ideally, I would like to have a common abstraction of trainer for more variants of GAN but it is not a very simple task. For the time being, we should have a `SequenceTrainer`.

(I also have thoughts of deprecating the `Trainer` in this release for an `ImageTrainer` and completely removing it in a later release)
